src/DataShare/DataShare/Account/RemoveAzDataShareAccount.cs

```diff
@@ -0,0 +1,133 @@
+        /// <summary>
+        /// The resource group name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "The resource group name of azure data share account.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "Azure data share account name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The resourceId of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource id of the azure data share account.",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ResourceIdCompleter(ResourceTypes.Account)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Azure data share account object.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.ObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "The azure data share account object.")]
+        [ValidateNotNullOrEmpty]
+        public PSDataShareAccount InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Return object (if specified).")]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string resourceId = null;
+
+            if (this.ParameterSetName.Equals(
+                ParameterSetNames.ResourceIdParameterSet,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                resourceId = this.ResourceId;
+            }
+
+            if (this.ParameterSetName.Equals(ParameterSetNames.ObjectParameterSet, StringComparison.OrdinalIgnoreCase))
+            {
+                resourceId = this.InputObject.Id;
+            }
+
+            if (!string.IsNullOrEmpty(resourceId))
+            {
+                var parsedResourceId = new ResourceIdentifier(resourceId);
+                this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                this.Name = parsedResourceId.GetAccountName();
+            }
+
+            this.ConfirmAction(
```

Comment:
@gaurangisaxena as [mentioned previously](https://github.com/Azure/azure-powershell/pull/9589#discussion_r302329874), this should be `ShouldProcess` instead of `ConfirmAction`.

Reply:
I was told that I could use either confirm action or should process. I chose to go with confirm action. Is that not okay?

Reply:
Misread the linked comment, the currently implementation is correct.